WIP - Initial pass to add namespaces to DeployCache

bookwyrm opened the pull request:
So that different deployers can have different deploy lists or a single deployer can have multiple deploy lists (production, staging, dry-run, etc).

I was running into a need for this in my development of the GitHub add-on (wanting a dry-run feature) so I went ahead and coded up a sample implementation. At the very least this gives us a possible implementation that we can kick around and discuss.

bookwyrm committed:
Initial pass to add namespaces to DeployCache

So that different deployers can have different deploy lists or a single deployer can have multiple deploy lists (production, staging, dry-run, etc).

bookwyrm commented:
I recognize that this causes issues with the Deploy Cache display in the wp-admin pages - would we have separate lines for each namespace?

leonstafford commented:
Looks good. Doesn't block trying the individual Add-ons view I want to look at next. 

DeployCache display - maybe just some links at top of page and extra param for WP-CLI commands that deal with it?

bookwyrm committed:
Merge branch 'deploy-cache-namespaces'

bookwyrm committed:
Need a compound primary key

So that the same path hash can be in the table multiple times for different namespaces

bookwyrm committed:
Merge remote-tracking branch 'upstream/master'

bookwyrm committed:
Merge branch 'master' into deploy-cache-namespaces

bookwyrm committed:
Need to take namespace into account for insert

Since it needs to be an argument to ON DUPLICATE KEY UPDATE so that we find and update the correct row.

Reformatted prepare values for better readability

bookwyrm committed:
Add support to show deploy cache info by namespace

Hidden behind a filter so that default plugin behavior hides the complexity of deploy cache namespaces and it’s only enabled when an add-on plugin needs it.

bookwyrm committed:
Change param to deploy_namespace

For better clarity of intent

bookwyrm committed:
Support showing paths by namespace

bookwyrm committed:
Update truncate to support namespace

bookwyrm committed:
Truncate by namespace if supplied
